### PR TITLE
fix(passwordbox): Fixed iOS auto-fill interfering with PasswordBox input

### LINKED ISSUE #7387: [iOS] PasswordBox does not work properly when iOS autofill feature is turned on.
### Current behavior

Tapping the `PasswordBox` brings up the keyboard, but as soon as a key is pressed, a "Strong Password" overlay that indicates iOS has auto-filled the box with a password shows up and the password box does not actually work, i.e. pressing the keys doesn't do anything and no value is stored in the `Password` property/binding:

image

It only appears on pages with 2 password boxes, which triggers the auto-fill feature - the login page that only contains a single password box works fine.

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

Place 2 password boxes on a page and try to enter values.

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

- Xcode 13 / iOS 15 SDK / Xamarin.iOS 15.xx SDK
- Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

`v3.11.0-dev.210`, `v3.11.0-dev.481`

### Affected platforms

iOS

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by mikernet**
GitHub Issue (If applicable): closes #7387

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

When password auto-fill is triggered on iOS it breaks PasswordBox input with the soft keyboard.

## What is the new behavior?

Auto-fill is disabled on PasswordBox.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxView.iOS.cs [hunk 1]
```diff
@@ -39,10 +40,20 @@ public override bool SecureTextEntry
...
+					{
+						// Disable auto-fill for now, does not work properly. The auto-filled value never becomes available on base.Text and blocks input on the soft keyboard.
+						if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
+						{
+							base.TextContentType = UITextContentType.OneTimeCode;
+						}
+						else if (UIDevice.CurrentDevice.CheckSystemVersion(10, 0))
+						{
+							base.TextContentType = NSString.Empty;
+						}
```
[the remark this answers is not in the document]
Comment (mikernet) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxView.iOS.cs:52]: `OneTimeCode` was introduced in iOS 12 and setting the `TextContentType` to `Empty` does not work anymore in that version. `TextContentType` was introduced in iOS 10.

As for unsetting - I don't think it matters. From what I can tell the property is not used anywhere else and only PasswordBox ever sets SecureTextEntry. It shouldn't affect anything about how PasswordBox works when revealed.

[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxView.iOS.cs:52]: Thanks!

**Review by jeromelaban — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**